src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -413,6 +427,68 @@ CHIP_ERROR DefaultDACVerifier::VerifyNodeOperationalCSRInformation(const ByteSpa
     return CHIP_NO_ERROR;
 }
 
+bool CsaCdKeysTrustStore::IsCdTestKey(const ByteSpan & kid) const
+{
+    return kid.data_equal(ByteSpan{ gTestCdPubkeyKid });
+}
+
+CHIP_ERROR CsaCdKeysTrustStore::AddTrustedKey(const ByteSpan & kid, const Crypto::P256PublicKey & pubKey)
+{
+    ReturnErrorCodeIf(kid.size() > SingleKeyEntry::kMaxKidSize, CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorCodeIf(kid.empty(), CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorCodeIf(mNumTrustedKeys == kMaxNumTrustedKeys, CHIP_ERROR_NO_MEMORY);
+
+    auto & entry = mTrustedKeys[mNumTrustedKeys];
+
+    entry.kidSize = kid.size();
+    memcpy(&entry.kidBuffer[0], kid.data(), kid.size());
+    entry.publicKey = pubKey;
+
+    ++mNumTrustedKeys;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CsaCdKeysTrustStore::AddTrustedKey(const ByteSpan & derCertBytes)
+{
+    // TODO: Verify cert against CD root of trust (i.e. verify signatures). To do so,
```

Comment:
Interesting that we do the check here but not in the `AddTrustedKey(const ByteSpan & kid, const Crypto::P256PublicKey & pubKey)` method above. Is that a loophole to bypass security?